src/ucp/proto/proto_am.inl

```diff
@@ -174,6 +174,8 @@ ucs_status_t ucp_do_am_zcopy_single(uct_pending_req_t *self, uint8_t am_id,
         req->send.state = saved_state; /* need to restore the offsets state */
         return status;
     } else {
+        ++req->send.uct_comp.count;
```

Comment:
In case of multi-thread offset+=len, posting of the operation and even restore of the state must be locked inside the same critical section to avoid using of wrong offset value, right? So, ++comp.count looks not significant increase of the locked piece of code.